### PR TITLE
switch to a snakefile and update taxonomy handling so no dropped prot accs

### PR DESCRIPTION
**Pull request by taylorreiter**
In #2, I documented the first way I made the clustered nr db. In that file, I used the `prot.accession2taxid` file to add taxids. However, I noticed that 60M accessions didn't have taxids...which is a lot. After some sleuthing, I realized there's a `prot.accession2taxid.FULL` which has all of the protein accessions documented, not just the ones with gi numbers. sigh. I tried to use the same recipe as presented in #2 to make the taxonomy sheet, but it ran out of ram bc the FULL sheet was over 2x bigger than the not full one. So i worked with the parts that ncbi provides for this. Through a series of stumbles, I had to run this code a few times and got sick of coordinating it myself and just put it in a snakefile.

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    wget -O {output} https://ftp.ncbi.nlm.nih.gov/blast/db/v5/FASTA/nr.gz # run on 03/13/2023 with db last updated 2023-03-11 21:09
+    '''
+
+rule cluster_nr_with_mmseqs:
+    input: "inputs/nr.gz"
+    output:
+        "nr_cluster.tsv",
+        "nr_rep_seq.fasta"
+    conda: "envs/mmseqs2.yml"
+    threads: 63
```
Comment (mertcelebi) [inline Snakefile:26]: is there a way to make this dynamic? perhaps doesn't matter, but curious

File: README.md [hunk 2]
```diff
@@ -4,4 +4,33 @@ NCBI now provides [a clustered nr database](https://ncbiinsights.ncbi.nlm.nih.go
 We were interested in using this database to reduce search times and to increase the taxonomic diversity of returned sequences when doing BLAST searches.
 However, as of March 2023, the database is not available for download.
 Therefore, we re-made this database ourselves.
-The [README.sh](./README.sh) file in this repository documents how we performed the clustering and created a taxonomy sheet that annotates the lowest common ancestor for each protein cluster.
+The [Snakefile](./Snakefile) in this repository documents how we performed the clustering and created a taxonomy sheet that annotates the lowest common ancestor for each protein cluster.
```
Comment (mertcelebi) [inline README.md:7]: I'd add a note to this README about, how much data this file downloads from where and the final product (a SQLite DB).

I'd also add a note about on which type of environment you ran this + download instructions for Taxonkit, given it doesn't exist on Conda yet.

**Review by mertcelebi — APPROVED** [review]
Approving with a comment. 

Note for a future discussion: I'd like to think more about how we could have avoided some of the repeat runs of this pipeline and any learnings from when it would've made sense to turn it into a Snakemake (or nf) pipeline.

File: Snakefile (hunk 1 above)
Comment (taylorreiter) [reply Snakefile:26]: Yes! 

https://stackoverflow.com/questions/48424357/how-to-set-threads-based-on-number-of-cores-available

**Pull request merged**